pandas/tests/indexes/test_interval.py

```diff
@@ -348,9 +355,118 @@ def test_get_loc_value(self):
         idx = IntervalIndex.from_arrays([0, 2], [1, 3])
         pytest.raises(KeyError, idx.get_loc, 1.5)
 
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_get_loc_value_closed_right_updated_behavior(self):
+
+        right = IntervalIndex.from_tuples([(0, 1), (2, 3)], closed='right')
+
+        pytest.raises(KeyError, right.get_loc, -0.5)
+        pytest.raises(KeyError, right.get_loc, 0)
+        assert right.get_loc(0.5) == 0
+        assert right.get_loc(1)   == 0
+        pytest.raises(KeyError, right.get_loc, 1.5)
+        pytest.raises(KeyError, right.get_loc, 2)
+        assert right.get_loc(2.5) == 1
+        assert right.get_loc(3)   == 1
+        pytest.raises(KeyError, right.get_loc, 3.5)
+
+        assert right.get_loc(Interval(0, 1, closed='right')) == 0
+        pytest.raises(KeyError, right.get_loc, Interval(1, 2, closed='right'))
+        assert right.get_loc(Interval(2, 3, closed='right')) == 1
+        pytest.raises(KeyError, right.get_loc, Interval(3, 4, closed='right'))
+        pytest.raises(KeyError, right.get_loc, Interval(0, 2, closed='right'))
+        pytest.raises(KeyError, right.get_loc, Interval(2.5, 3, closed='right'))
+
+        pytest.raises(KeyError, right.get_loc, Interval(0, 1, closed='left'))
+        pytest.raises(KeyError, right.get_loc, Interval(1, 2, closed='left'))
+        pytest.raises(KeyError, right.get_loc, Interval(2, 3, closed='left'))
+        pytest.raises(KeyError, right.get_loc, Interval(3, 4, closed='left'))
+        pytest.raises(KeyError, right.get_loc, Interval(0, 2, closed='left'))
+        pytest.raises(KeyError, right.get_loc, Interval(2.5, 3, closed='left'))
+
+        pytest.raises(KeyError, right.get_loc, Interval(0, 1, closed='both'))
+        pytest.raises(KeyError, right.get_loc, Interval(1, 2, closed='both'))
+        pytest.raises(KeyError, right.get_loc, Interval(2, 3, closed='both'))
+        pytest.raises(KeyError, right.get_loc, Interval(3, 4, closed='both'))
+        pytest.raises(KeyError, right.get_loc, Interval(0, 2, closed='both'))
+        pytest.raises(KeyError, right.get_loc, Interval(2.5, 3, closed='both'))
+
```

Comment:
Can you add here (and in the ones below) a test for the case of indexing with an Interval that is larger (fully contains the ones in the index), and that this raises as well? (to cover your `x.loc[pd.Interval(-10,10)]` which is currently working from the issue https://github.com/pandas-dev/pandas/issues/16316#issuecomment-300544957)

So eg like `pytest.raises(KeyError, right.get_loc, Interval(-10, 10, closed='both'))`